[READY] Space Ninja Rework

IndieanaJones opened the pull request:
## About The Pull Request

This PR completely reworks the midround antagonist known as Space Ninja.  There is a lot of stuff to go over in this one, so we might as well get started.  Grab a drink, get cozy, because class is in session now.

**WHAT WAS WRONG WITH SPACE NINJA**

Hoo boy.  Let me tell you, it'd probably be easier to list everything NOT wrong with space ninja.

- Insane power.  Space ninja was extremely strong power-wise and could easily evade capture and kill targets at his leisure.  A space ninja that knew what tools he had could pretty much fight the entire station and win due to how many things he had going for him.

- Shitty objectives.  Yeah, I sure do love it when a midround antagonist who has the power of god and anime on his side can just teleport to my location and kill me with no chance of me retaliating, yep.  Bonus points if it was a Nanotrasen ninja and you lost your antag to it.

- Old old old old old.  Space ninja was extremely old, and nobody has ever really taken time to change his core mechanics, only make sure he functioned to a degree.  That wouldn't be so bad if space ninja wasn't a feature from over a decade ago.

- Along with being old conceptually, space ninja has sort of become moldy in the code.  It's really old, OK?  And while people have messed with it on occasion, nobody has actually done a complete cleaning of it.  Along with this, there's parts of code that should be the ninja module that aren't, some parts of code that shouldn't be in the ninja module but are, it's a terrible mess, let me tell you.  Going in blind and trying to understand what everything did at a glance was a terrible experience, and the more I delved, the more completely undocumented shit I found.  It was a treasure trove, I tell you, one filled with hidden features and old developers talking to each other in the comments, really felt like a time capsule.

- Overloaded kit.  Space ninja had a lot of tools which only heightened his status as a god-tier murderboner, especially since a lot of them didn't have any drawbacks to space ninja and could easily overwhelm the crew.  The amount of get-out-of-jail free cards was insane, and the power costs for some of his stronger abilities were very, very low.

- Had gay baby jail.  Space ninja had a net that could capture targets and send them to gay baby jail, which is essentially like instantly killing someone with no chance of escape, except they don't even get the pleasure of dying afterwards.  Believe me, I don't think there's a single human being who could unironically say that this was a fun mechanic.  Not one.

- Randomized stats.  This one wasn't even on the wiki, but turns out that space ninja had randomized suit stats which would affect various things in their kit.  Pretty much what that meant was if you were unlucky, your space ninja could be a space dud and the game would just leave you to find that out on your own.  Great game design.

- Old antag classic balancing strat.  Blatantly overpowered?  Just make it extremely rare.  While that's not horrible, we could always use more variety for the 1 hour shifts we aim for instead of the theoretical "what-if" 5-hour shift.

Okay, with everything that was wrong being out of the way and taking more words to explain than most PRs have in general, let's talk about how space ninja has changed to fit in /tg/station in the year 2020.

**WHAT CHANGED**

Long story short, a lot changed.  I'm going to try and document as much as I can here, but if you see something I missed, do please tell me and I'll add it here.  The concept being aimed for is to take space ninja and make him an antagonist with no much upright killing power, but a pretty good stealth and sabotage game.  So, what this means is to expect the man-sideways numbers to go down and adjustments to be made around the idea of space ninja being a stealth antag with sabotage objectives.  Alright then, I guess the best place to start is the ninja's new kit of powers.  I'll tell you about all the changes I made to all his tools, but first, let me list everything that got removed entirely:

- Smoke bombs
- X4 charge
- Jetpack

Those features/items are just gone entirely, not coming back.  Everything else is still included, albeit likely in a heavily altered state.  Let's begin:

**THE KATANA**
- Damage decreased from 40 to 30
- Throw damage increased from 25 to 30
- Teleport recharge speed increased from 150 to 200 (takes 33% longer to recharge)
- Can no longer teleport into solid objects.  Yes, that means you can't phase into doors and walls anymore, but if you can see the area behind the door, you can just teleport there instead.  Consider it a rule of thumb that if you can see it and you can target a non-dense object on that tile, you can teleport there.  Simple, right?
- No longer emags objects it hits.  Pretty sure this was an undocumented one, but I guess that's why we have a spider clan cyborg lawset.

If being unable to pass through locked doors scares you now, just keep reading, you'll see why those changes were made a bit (or a lot, this might take a while).

**THE X4 CHARGE'S REPLACEMENT**

Yeah, so the X4 charge is gone, but its been replaced with something called a spider charge.  It's a more powerful, unfocused bomb meant to deal some hull damage in the area it detonates.  However, this bomb can't be detonated anywhere, but only in a specific location tied to one of your new objectives, which we'll get into later.

**ALL ARMOR CHANGES**
- The protective armor has been nerfed down from mining-gear numbers to more reasonable numbers.  The new numbers are
armor = list(MELEE = 40, BULLET = 30, LASER = 20,ENERGY = 30, BOMB = 30, BIO = 30, RAD = 30, FIRE = 100, ACID = 100)
and I can guarantee you that it's a nerf from what it was.  Old numbers were armor = list(MELEE = 60, BULLET = 50, LASER = 30,ENERGY = 15, BOMB = 30, BIO = 30, RAD = 30, FIRE = 100, ACID = 100), so Melee, Bullet, and Laser were all nerfed down some.

**COWL**
- Armor nerf applied

**SHOES**
- Armor nerf applied

**THE ACTUAL SUIT ITSELF**
- Armor nerf applied
- No longer randomizes power stats
- Passive diagnostics on the Status screen removed
- Female sprite is now determined on body type, not gender.  NipLEDs are for everyone as long as you have the right body type.
- Now gives the user the anti-guns trait while the suit is on.

**POWER CHANGES FROM HERE UNTIL THE BALANCE PATCHNOTES ARE OVER**

**ADRENALINE BOOST**
- Max capped to one adrenaline boost stored in the suit at any time
- Always takes 20 radium to refill the suit with another adrenaline boost.  Injects 5 radium into you when it ends.

**NINJA NET**
- Power cost increased to take 40% of the base cell's max charge (40E)
- Net no longer gay baby jails you period
- Durability greatly increased
The idea here is that you can use the net in a dire 1v1 situation, but you can't just go spamming it at everyone.  If its still too annoying, it'd be the first thing I'd be down to remove, I'm just hesitant since the net sort of is iconic at this point.

**EMP**
- Power cost doubled from 25E to 50E (25% of the base cell's power to 50%)
Considering the flavor of the ninja, an EMP is fitting and will help with an objective we'll talk about later.

**GLOVES**
- Carbon stun turned into a knockdown which last 3 seconds, costs the same as it did.

**NINJA STARS**
- No longer deal 30 throw damage, instead being based around the stamina throwing stars.  What this means is they do little damage on hit, but deal stamina damage as the target walks around with them still embedded.  Good for forcing unarmored targets to flee or try to finish you off.

**SUIT DIAGNOSTICS**
This is actually a brand new ability which just puts what information the suit used to display on your status page into chat instead.  Why didn't I just leave it, you ask?  The code for displaying the info was in human.dm, and that's bad practice, so since it didn't really need to be removed, here it is as an ability instead.

**SWORD RECALL**
- No changes

**STEALTH**
- Increased the potency of the invisibility
- If attacked while invisible, disables invisibility and all suit powers for 5 seconds

Considering that SN is a stealth antag now, it made sense to buff the ability to actually enable you to stealth.  You're pretty hard to see now, and any tile with some noise on it makes a great hiding spot for you.  However, there's actually a punishment now with getting caught while invisible, so try not to get caught, simply put.

**GLOVE TOGGLE**
- Now is an ability of the gloves instead of the suit.  Still can't access the special interactions without the suit, though.

Speaking of the gloves, you'll notice they aren't listed here.  That's because they got changes, but they're all tied into the objectives, so let's discuss the new objectives now.

**NEW OBJECTIVES**
Space Ninja gets 7 objectives upon spawning in.  Their types are guaranteed, but rest be assured they have variation in them.  Let's just go down the list and talk about them.

**GENERAL CHANGES**
- No longer has a chance to be Nanotrasen-sided.  You know, the Spider Clan was looking to get rid of the space ninja program, all that's kept it afloat was Syndicate funding after all this time.  So no more Nanotrasen ninjas, all ninjas are antagonists against the crew.

FIRST OBJECTIVE - "Use your gloves to convert a cyborg to aide you in sabotaging the station."

So, this is the first of new abilities the gloves have picked up.  If you can get your hands on a cyborg for 6 seconds, you'll pretty much emag them but in a much more special way.  Instead of just becoming an emagged cyborg, they actually turn into one of the three syndicate modules that you would know from Nuclear Operatives.  Of course, this makes them very strong, but note that you can only do this once and you and cyborg don't particularly mesh.  The idea is that the cyborg is just meant to cause as much general chaos as possible so you and any other antagonists can get their jobs done while they're distracted dealing with the very obvious cyborg threat in front of them.  Considering they're working alone, it puts them in a much more vulnerable state than during Nuke Ops, and also gives them some time to shine outside of the Nuke Ops umbrella.  Why can a Spider Clan agent make a syndicate cyborg, you ask?  Well, they funded it the project, so it'd make sense that they wouldn't mind giving out some of their tech, especially if it hurts Nanotransen.

The second objective is the stealing research nodes objective Space Ninja had prior to this rework.  So while its not technically new, it fits in with the stealth and sabotage aspect, and so can stay.

THIRD OBJECTIVE - "Use your gloves to doorjack [doorobjective.doors_required] airlocks on the station."

Another new function of your gloves lets you pretty much become a walking greytide virus.  Slap a door with your bad boys and watch as any door of your dreams (minus bolted ones) opens and then fries.  You'll be expected to do this to around 20-40 doors, so you can either go very proactive (and immediately cause suspicion) or just do it as you need to over the course of your existence. 

FOURTH OBJECTIVE - "Detonate your starter bomb in [bombobjective.detonation_location].  Note that the bomb will not work anywhere else!"

Remember the spider charge from up top?  Yeah, here is where it comes into play.  You'll be expected to use that bad boy on a random part of the station, and you can't set it up anywhere else.  Completion is awarded when the bomb detonates, so make sure it actually goes off!

FIFTH OBJECTIVE - "Use your gloves on a security console to set everyone to arrest at least once.  Note that the AI will be alerted once you begin!"

Your gloves can now set everyone in the security records to arrest.  However, this action takes about 20 seconds to complete, and the AI will be notified the moment you start, so plan accordingly.

SIXTH OBJECTIVE - "Use your gloves on a communication console in order to bring another threat to the station.  Note that the AI will be alerted once you begin!"

You can use your gloves on a communication console once to start either the swarmer or space pirate event.  This takes 30 seconds to complete the AI is alerted once you start, so make sure nobody is around to stop you!  Once you complete this objective, you'll be given additional allies in your war of terror on the station, so its best suggested that you try to assist them as much as possible.

The seventh objective is just the survival one, nothing fancy, but if you pulled off all the other stunts you were required to do, don't be surprised if everyone is trying to hunt you down!

So, that's all the gameplay changes (I think).  It's a lot of change, and though I can test it on singleplayer, determining how it works out on live will obviously be something that a testmerge will require.  I'm very open to any suggestions anyone wants to make, so please do comment below on your thoughts.  With everything gameplay related out of the way, its time for for other changes:

- Event file moved into the events folder
- Moved the suit initialization file into the equipment powers folder
- Renamed the antagonist folder to space_ninja, along with the file inside
- Removed the space ninja suit code in human.dm
- Rigged Space Ninja up to the Dynamic midround system
- Buffed Space Ninja's spawn chance to be akin to Blob and Nightmare
- Moved the hacking notification for machinery out of the ninja files and into _machinery.dm where it belongs
- Autodoc'd all the code
- Reorganized the file layouts, got rid of some old commentary.  Tried to keep the idea of all new procs being below any pre-established ones.
- Removed the ninja file in the actions folder, all relevant action info is now stored in the respective power's file
- Updated the one_click_antag() admin verb to work with the new ninja
- Replaced obscure variable names with more obvious ones

If you want a summary of all that, pretty much what I did is make it so everything is much easier to read and if need be, to remove ninja all you'd need to do is to delete the file and the antagonist and event files.  Also, ninja now spawns more commonly like a regular midround antagonist since he isn't god anymore.

## Why It's Good For The Game

I don't think words can describe how awful Space Ninja up until this point was.  Ask anyone and they'll agree that as he is right now, he's bullshit to fight not fun at all to go up against.  Simply put, he was a code artifact so awful that nobody wanted to touch him and he even outlived his supposed-to-be successor to the point that the successor doesn't even work anymore.  As a person who focuses a lot of my work on tryng to bring all of our current midround antagonists to a good level of balance and fun, it's blatantly clear that change needed to happen, and this is it.

## Changelog
:cl:
balance: The Spider Clan has recently taken up the Space Ninja project again along with the Syndicate.  Space Ninjas have been drastically changed as a result, becoming much weaker and more stealth oriented.  As a result of cutting costs per ninja, more ninjas were able to be hired.  Expect to see them around more often.
/:cl:

IndieanaJones committed:
Move suit_initialization into better location

IndieanaJones committed:
Clean up cost check

IndieanaJones committed:
Clean up net using

IndieanaJones committed:
Clean up Ninja Star

IndieanaJones committed:
Clean Up Ninja Stealth

IndieanaJones committed:
Clean Up Sword Recall

IndieanaJones committed:
Move the event file

IndieanaJones committed:
Re-Merge the Changes

IndieanaJones committed:
Get your ninja code out of my human.dm

IndieanaJones committed:
Move the event into events folder and clean it up

IndieanaJones committed:
Can the old space ninja antagonist file

IndieanaJones committed:
Roll in the shiny new bad boy

IndieanaJones committed:
Get rid of actions/ninja.dm and move its info into respective ability files

IndieanaJones committed:
Update one_click_antag with new ninja

IndieanaJones committed:
Move proc out of ninja code into proper location

IndieanaJones committed:
Update the dme properly this time

IndieanaJones commented:
Sorry this a long one, but a lot of stuff was changed.  There's a few grammar issues here and there since I wrote the description at 2 AM, but if you have any questions, please ask.

ATH1909 commented:
Note that you can currently bash a ninja net that you're inside of with an object a few times to destroy it, IIRC; it's not a gg no re unless you're already in crit or stunned.

ReplicantTerry commented:
pretty cool tbh

IndieanaJones commented:
> Note that you can currently bash a ninja net that you're inside of with an object a few times to destroy it, IIRC; it's not a gg no re unless you're already in crit or stunned.

While that's true, getting sent to the dimension of no return is still not very fun and I don't think it should be included.

ATH1909 commented:
> > Note that you can currently bash a ninja net that you're inside of with an object a few times to destroy it, IIRC; it's not a gg no re unless you're already in crit or stunned.
> 
> While that's true, getting sent to the dimension of no return is still not very fun and I don't think it should be included.

If you're acknowledging that that's true, and the ninja net no longer sends you to the gbj dimension, why is its price so high?

IndieanaJones commented:
> > > Note that you can currently bash a ninja net that you're inside of with an object a few times to destroy it, IIRC; it's not a gg no re unless you're already in crit or stunned.
> > While that's true, getting sent to the dimension of no return is still not very fun and I don't think it should be included. 
> If you're acknowledging that that's true, and the ninja net no longer sends you to the gbj dimension, why is its price so high?

Because it immobilizes a single target completely which is pretty powerful and I upped the durability on it.

ATH1909 commented:
Not sure I'm a fan of the syndicate cyborg conversion. The assault and medical cyborgs really weren't designed to operate alone, as they have no way to heal themselves without the ninja's assistance (aside from, say, emagging an engiborg and getting them to welder you, but borg vs. borg melee is a coinflip). It's also a bit sad that the ability to emag stuff with your katana was removed.

What if the spider clan charge summoned the syndicate drop pod (from the nuke op base) instead of making an explosion? The drop pod could contain ghost-controlled syndicate cyborgs (one of each type?) or something when it comes down. You could then give the ninja's gloves the ability to emag stuff, allowing him to subvert borgs by emagging them if he wants to.

ATH1909 commented:
What if you kept the GBJ, but made it so that when the ninja dies/explodes, a two-way portal opens up that can transport people between ninja jail and where the ninja died?

ATH1909 commented:
I also feel like the 25 force of the ninja's energy katana is a bit... underwhelming. Could you maybe buff it up to esword damage (30)?

SkeletalElite commented:
Metal Gear Quote removal is lame. Leave the metal gear quotes

IndieanaJones commented:
I'd be willing to buff up the katana up to 30 and re-add the metal gear solid lines back in, but I don't want to give the gloves full emag powers nor buff the current cyborg scenario (honestly thought people would claim it'd be too OP instead of too weak).  Assault gets some serious firepower for something outside of a nukie round, medic can heal the ninja (since the suit is a full set of armor, you can't use most common healing applications without taking your suit off), and saboteur can...  sabotage.  I'd rather see how it plays out in-game before buffing it.

Tlaltecuhtli commented:
can you give it a mimetic power like turn into a potted plant?

IndieanaJones commented:
> can you give it a mimetic power like turn into a potted plant?

Considering that you can just pick up a potted plant to disguise with it, along with the fact that the invisibility power is already in, I wouldn't deem a feature like this necessary.

Fikou commented on code/modules/ninja/suit/ninja_equipment_actions/ninja_suit_initialisation.dm:
```diff
@@ -24,6 +24,9 @@ GLOBAL_LIST_INIT(ninja_deinitialize_messages, list(
 ))
 
+/datum/action/item_action/initialize_ninja_suit
+	name = "Toggle ninja suit"
```
```suggestion
	name = "Toggle Ninja Suit"
```

Fikou commented:
did you even code a way for him to doorjack

ATH1909 commented on code/modules/antagonists/space_ninja/space_ninja.dm:
```diff
@@ -0,0 +1,118 @@
+/datum/antagonist/ninja/proc/addMemories()
+	antag_memory += "I am an elite mercenary of the mighty Spider Clan. A <font color='red'><B>SPACE NINJA</B></font>!<br>"
+	antag_memory += "Surprise is my weapon. Shadows are my armor. Without them, I am nothing. (//initialize your suit by clicking the initialize UI button, to use abilities like stealth)!<br>"
+	antag_memory += "Officially, The Syndicate are my employer.<br>"
```
also, this should say "The Syndicate **is** my employer.", not "The Syndicate **are** my employer.".

itseasytosee commented:
If you want to remove the net you should probably just remove the net. I doubt many people will miss it.

IndieanaJones commented:
> did you even code a way for him to doorjack

It just uses the emag functionality, since its everything I wanted, really.

IndieanaJones committed:
Update code/modules/ninja/suit/ninja_equipment_actions/ninja_suit_initialisation.dm

Co-authored-by: Fikou <piotrbryla@onet.pl>

IndieanaJones committed:
Update code/modules/ninja/suit/gloves.dm

Co-authored-by: Fikou <piotrbryla@onet.pl>

IndieanaJones committed:
Update the antagonist file

IndieanaJones committed:
Update ninjaDrainAct

IndieanaJones committed:
Re-Add MGS Voice Lines

IndieanaJones committed:
Up the Katana Damage Back Up to 30

IndieanaJones commented:
Alright, first round of patch notes:
- katana force upped from 25 to 30
- Re-added the MGS references
- Removed Hyjack, added Hijack
- Removed the employer line

CRITAWAKETS commented:
**TENNO SKOOM!!**

Also, the ability to change a cyborg into a nukie assault cyborg is _definitively way too strong_ because the nukie assault borg has an _**UNLIMITED AMMO LMG**_

IndieanaJones commented:
Old invisibility:
![NinjaOldInvisibility](https://user-images.githubusercontent.com/47086570/92334385-3e87d180-f05b-11ea-87f9-ca69e11d60d9.png)
New Invisibility:
![NinjaNewInvisibility](https://user-images.githubusercontent.com/47086570/92334388-434c8580-f05b-11ea-890f-1cc48fe407c8.png)

IndieanaJones commented:
> Also, the ability to change a cyborg into a nukie assault cyborg is _definitively way too strong_ because the nukie assault borg has an _**UNLIMITED AMMO LMG**_

While it is an extremely strong weapon, note that a single well-placed flash, EMP, electronic gun, or the other borg disablers will guarantee its death.  It also can't heal unless the ninja gets a welder or someone else agrees to to do it, so having some power in its corner isn't awful.

necromanceranne commented:
I like the proposals so far but I have just one minor thing to bring up. Did you test just how much damage 20 radium will do to you? Because I suspect that just sets you to lethal dosages of radiation and you die as a result. And given ninjas have few means of curing that easily, let alone the crew, I'm not sure why even include this over giving them a dose of bad stims instead, which has all the relevant downsides upfront.

Additionally, I don't exactly think the full suite of nukie borgs is a good idea, but a saboteur borg is actually perfect for a stealth ninja. That borg can help you set up the station for disaster, while less likely to cause issues like 'Assault borg murders everyone in a hail of bullets and grenades'.

IndieanaJones commented:
> I like the proposals so far but I have just one minor thing to bring up. Did you test just how much damage 20 radium will do to you? Because I suspect that just sets you to lethal dosages of radiation and you die as a result. And given ninjas have few means of curing that easily, let alone the crew, I'm not sure why even include this over giving them a dose of bad stims instead, which has all the relevant downsides upfront.

Yeah, it doesn't really do anything to him.  It used to inject a random amount of radium which had the ability to inject way more as well, I just set it to be a constant 20.  I've injected myself a bunch of times during testing and I never keeled over.  Maybe it has to do with the suit?

> Additionally, I don't exactly think the full suite of nukie borgs is a good idea, but a saboteur borg is actually perfect for a stealth ninja. That borg can help you set up the station for disaster, while less likely to cause issues like 'Assault borg murders everyone in a hail of bullets and grenades' or 'Medical borg does nothing and seems like a horrible choice'.

I understand the concerns, but I want to see it in action before removing it or tweaking it.  We have testmerges for this purpose, so I want to see how it actually plays out in a live environment before ditching the concept entirely.

GravitronBlue commented on code/modules/ninja/suit/ninjaDrainAct.dm:
```diff
@@ -1,263 +1,321 @@
 				break
 
-
-//CARBON MOBS//
-/mob/living/carbon/ninjadrain_act(obj/item/clothing/suit/space/space_ninja/S, mob/living/carbon/human/H, obj/item/clothing/gloves/space_ninja/G)
-	if(!S || !H || !G)
-		return INVALID_DRAIN
-
-	. = DRAIN_MOB_SHOCK_FAILED
-
-	//Default cell = 10,000 charge, 10,000/1000 = 10 uses without charging/upgrading
-	if(S.cell && S.cell.charge && S.cell.use(1000))
-		. = DRAIN_MOB_SHOCK
-		//Got that electric touch
-		var/datum/effect_system/spark_spread/spark_system = new /datum/effect_system/spark_spread()
-		spark_system.set_up(5, 0, loc)
-		playsound(src, "sparks", 50, TRUE)
-		visible_message("<span class='danger'>[H] electrocutes [src] with [H.p_their()] touch!</span>", "<span class='userdanger'>[H] electrocutes you with [H.p_their()] touch!</span>")
-		electrocute_act(25, H)
```
Since I don't see any documentation on it, are the stungloves aspect of the ninja suit being removed or did it get rebuilt elsewhere in the code?
Assuming complete removal I ask whether the `DRAIN_MOB_SHOCK` and 'DRAIN_MOB_SHOCK_FAILED' defines over in __ninjaDefines.dm are still used else and still worth keeping.

Code stuff aside, I figured ninja gloves would see a hit eventually given the current views on one-click-stuns, yet would it still be reasonable to rework it to share stunprod mechanics and take two shocks to incapacitate someone as space ninjas have little else in the way of non-lethal takedowns.

Similar thoughts for the energy net jail and whether it could return abductees much like those taken by syndicate contractors.

IndieanaJones replied:
I don't believe DRAIN_MOB_SHOCK or DRAIN_MOB_SHOCK_FAILED are used anywhere, I'll look to remove them later.  As of right now, you still do have two options of non-lethal takedown, the net and the new ninja stars, which drain stamina once attached to their target and they move around.  I will consider having the net work like the contractor's capture, but I'm still inclined to keeping it out as it is right now.

Tharcoonvagh commented:
_did you just remove the "nothing" objective_

Joke aside ninja needed some balancing, thanks dude

anconfuzedrock commented:
I don't even care what he did, changing ninja is the most based thing I've ever heard of
removing nothing obj is sad tho

Rohesie commented on code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm:
```diff
@@ -551,3 +551,45 @@
+	var/list/spawn_locs = list()
+
+/datum/dynamic_ruleset/midround/from_ghosts/space_ninja/execute()
+	for(var/obj/effect/landmark/carpspawn/C in GLOB.landmarks_list)
+		spawn_locs += (C.loc)
```
```suggestion
	for(var/obj/effect/landmark/carpspawn/carp_spawn in GLOB.landmarks_list)
		spawn_locs += (carp_spawn.loc)
```

Rohesie commented on code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm:
```diff
@@ -551,3 +551,45 @@
+	if(!spawn_locs.len)
+		message_admins("No valid spawn locations found, aborting...")
+		return MAP_ERROR
+	. = ..()
```
```suggestion
	return ..()
```

Rohesie commented on code/modules/antagonists/space_ninja/space_ninja.dm:
```diff
@@ -0,0 +1,116 @@
+	var/give_equipment = TRUE
+
+/datum/antagonist/ninja/apply_innate_effects(mob/living/mob_override)
+	var/mob/living/M = mob_override || owner.current
```
I'm not going to go through every single upper-case-letter-named variable, as I've done it aplenty on other PRs of yours.
Please fix those with informative variable names all around.

Rohesie commented on code/modules/antagonists/space_ninja/space_ninja.dm:
```diff
@@ -0,0 +1,116 @@
+	addMemories()
+	if(give_equipment)
+		equip_space_ninja(owner.current)
+	. = ..()
```
`return ..()` at the end

Rohesie commented on code/modules/ninja/suit/suit.dm:
```diff
@@ -96,59 +107,108 @@ Contents:
+	if(s_coold != 0)
+		to_chat(user, "<span class='warning'><b>ERROR</b>: suit is on cooldown.</span>")
+		return FALSE
+	if(istype(action, /datum/action/item_action/ninjastatus))
+		ninjastatus()
+		return TRUE
+	if(istype(action, /datum/action/item_action/ninjaboost))
+		ninjaboost()
+		return TRUE
+	if(istype(action, /datum/action/item_action/ninjapulse))
+		ninjapulse()
+		return TRUE
```
Do a redefined proc instead of a long istype check chain.

IndieanaJones replied:
How exactly am I to do a redefined proc chain for these?  They don't do anything but alert the suit when one of them has been pressed, and share no common proc as far as I'm aware.

Rohesie commented on code/modules/ninja/suit/suit.dm:
```diff
@@ -164,45 +224,15 @@ Contents:
+  * Can be called to entire rid of the suit pieces and the suit itself.
+  */
+/obj/item/clothing/suit/space/space_ninja/proc/terminate()
+	qdel(n_hood)
+	qdel(n_gloves)
+	qdel(n_shoes)
```
don't keep references to deleted thing, else you'll force a hard delete
null the variables
you have `QDEL_NULL` for that

ATH1909 commented:
> ninjas also trying to steal beeg power cells is kinda obnoxious and it might be better to make them unable to change the cell (but make it bigger)

They currently can't "downgrade" their cell, which makes using hypercharged cells or the like from xenobio a bit tricky.

Kaffe-work commented:
the fact that you force the ninja to do their objectives is pretty lame, no other solo antag "have" to do their goals to play optimally, the fact that you can only bomb one spot, hijack one Borg and so on is pretty lame.

cacogen commented:
This is really cool. Do xenomorphs next.

SkeletalElite commented:
> the fact that you force the ninja to do their objectives is pretty lame, no other solo antag "have" to do their goals to play optimally, I.e that you can only bomb one spot, hijack one Borg and so on is pretty lame.

Eh the item/function only exists for the objective. If you don't want to do the objective just throw the charge away. As for the borg thing, I don't know why you'd not want to do that objective almost any borg except a med borg is pretty helpful to you.

IndieanaJones commented:
So, I have a merge conflict I can't resolve in the web editor and I can't resolve it in my setup either.  Not sure what to do.

Kaffe-work commented:
perhaps if you could choose between three locations or so it'd be more interesting. I just feel as if it is railroading a bit much for my tastes.

IndieanaJones committed:
Attempting a Risky Manuever

IndieanaJones committed:
Adjusting some numbers to fit the delta time PR

IndieanaJones committed:
Fix Var Names and do a return ..()

IndieanaJones commented:
> the fact that you force the ninja to do their objectives is pretty lame, no other solo antag "have" to do their goals to play optimally, I.e that you can only bomb one spot, hijack one Borg and so on is pretty lame.

Space Dragon?  Blob?  Heretic?  I will consider giving a second location to plant the bomb at, but I don't feel compelled to right now.

IndieanaJones committed:
Remove Unused Defines, Update qdels to QDEL_NULLs

Rohesie commented on code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm:
```diff
@@ -551,3 +551,45 @@
+
+/datum/dynamic_ruleset/midround/from_ghosts/space_ninja/execute()
+	for(var/obj/effect/landmark/carpspawn/carp_spawn in GLOB.landmarks_list)
+		spawn_locs += (carp_spawn.loc)
```
```suggestion
		spawn_locs += carp_spawn.loc
```
Superfluous parentheses that don't really help with clarity.

Rohesie commented on code/modules/antagonists/space_ninja/space_ninja.dm:
```diff
@@ -0,0 +1,116 @@
+	
+	//Explosive plant, the bomb will register its completion on priming
+	var/datum/objective/plant_explosive/bombobjective = new /datum/objective/plant_explosive()
+	var/sanity = 0
+	while(sanity < 100)
+		var/area/selected_area = pick(GLOB.sortedAreas)
+		if(selected_area && is_station_level(selected_area.z) && (selected_area.area_flags & VALID_TERRITORY))
+			bombobjective.detonation_location = selected_area
+			break
+		sanity++
```
```suggestion
	for(var/sanity in 1 to 100) // 100 checks at most.
		var/area/selected_area = pick(GLOB.sortedAreas)
		if(selected_area && is_station_level(selected_area.z) && (selected_area.area_flags & VALID_TERRITORY))
			bombobjective.detonation_location = selected_area
			break
```

Rohesie commented on code/modules/antagonists/space_ninja/space_ninja.dm:
```diff
@@ -0,0 +1,116 @@
+	survival.owner = owner
+	objectives += survival
+
+/proc/remove_ninja(mob/living/ninja)
+	if(!ninja || !ninja.mind)
```
On which conditions would this proc be called without a parameter?

Rohesie commented on code/modules/antagonists/space_ninja/space_ninja.dm:
```diff
@@ -0,0 +1,116 @@
+	datum.on_removal()
+	return TRUE
+
+/proc/is_ninja(mob/living/possible_ninja)
+	return possible_ninja && possible_ninja.mind && possible_ninja.mind.has_antag_datum(/datum/antagonist/ninja)
```
Could you turn this into a define, along with the `istype` ones?

Rohesie commented on code/modules/antagonists/space_ninja/space_ninja.dm:
```diff
@@ -0,0 +1,116 @@
+	to_chat(owner.current, "I am an elite mercenary of the mighty Spider Clan. A <font color='red'><B>SPACE NINJA</B></font>!")
+	to_chat(owner.current, "Surprise is my weapon. Shadows are my armor. Without them, I am nothing. (//initialize your suit by right clicking on it, to use abilities like stealth)!")
+	owner.announce_objectives()
+	return
```
```suggestion

```

Rohesie commented on code/modules/events/space_ninja.dm:
```diff
@@ -0,0 +1,58 @@
+
+/datum/round_event/ghost_role/space_ninja/spawn_role()
+	var/list/spawn_locs = list()
+	for(var/obj/effect/landmark/carpspawn/C in GLOB.landmarks_list)
+		spawn_locs += (C.loc)
```
```suggestion
	for(var/obj/effect/landmark/carpspawn/carp_spawn in GLOB.landmarks_list)
		if(!isturf(carp_spawn.loc))
			stack_trace("Carp spawn found not on a turf: [carp_spawn.type] on [isnull(carp_spawn.loc) ? "null" : carp_spawn.loc.type]")
			continue
		spawn_locs += carp_spawn.loc
```

Rohesie commented on code/modules/events/space_ninja.dm:
```diff
@@ -0,0 +1,58 @@
+	var/key = selected_candidate.key
+
+	//Prepare ninja player mind
+	var/datum/mind/Mind = new /datum/mind(key)
+	Mind.assigned_role = ROLE_NINJA
+	Mind.special_role = ROLE_NINJA
+	Mind.active = TRUE
```
Could you set these up in `mind/Initialize()` and use lower-case named variables?

IndieanaJones replied:
What mind/Initialize() are you referring to?  I can't anything pre-existing to utilize, do you want me to make one?

IndieanaJones replied:
OK, so according to mind's documentation, we shouldn't be creating minds for something like this.  So, I've made it use the default mind every mob spawns with and then assign the roles on when the space ninja antag status is gained, which also means space ninjas spawned from unconventional means now gain the proper roles as well.

Rohesie commented on code/modules/events/space_ninja.dm:
```diff
@@ -0,0 +1,58 @@
+
+/proc/create_space_ninja(spawn_loc)
+	var/mob/living/carbon/human/new_ninja = new(spawn_loc)
+	var/datum/preferences/A = new()//Randomize appearance for the ninja.
+	A.real_name = "[pick(GLOB.ninja_titles)] [pick(GLOB.ninja_names)]"
+	A.copy_to(new_ninja)
```
What does `A` stand for? Can you use better variable names?

Rohesie commented on code/modules/ninja/energy_katana.dm:
```diff
@@ -59,16 +65,29 @@
 	..()
 
-/obj/item/energy_katana/proc/returnToOwner(mob/living/carbon/human/user, doSpark = 1, caught = 0)
+/obj/item/energy_katana/Destroy()
+	QDEL_NULL(spark_system)
+	return ..()
```
```suggestion
/obj/item/energy_katana/Destroy()
	QDEL_NULL(spark_system)
	QDEL_NULL(jaunt)
	return ..()
```

Rohesie commented on code/modules/ninja/ninja_explosive.dm:
```diff
@@ -0,0 +1,37 @@
+		to_chat(user, "<span class='notice'>This isn't the location you're supposed to use this!</span>")
+		return
+	detonator = user
+	. = ..()
```
```suggestion
	return ..()
```

Rohesie commented on code/modules/ninja/suit/ninja_equipment_actions/energy_net_nets.dm:
```diff
@@ -0,0 +1,47 @@
+			playsound(src, 'sound/weapons/slash.ogg', 80, TRUE)
+
+/obj/structure/energy_net/Destroy()
+	if(!QDELETED(affecting))
+		affecting.visible_message("<span class='notice'>[affecting.name] is recovered from the energy net!</span>", "<span class='notice'>You are recovered from the energy net!</span>", "<span class='hear'>You hear a grunt.</span>")
```
```suggestion
	if(!QDELETED(affecting))
		affecting.visible_message("<span class='notice'>[affecting.name] is recovered from the energy net!</span>", "<span class='notice'>You are recovered from the energy net!</span>", "<span class='hear'>You hear a grunt.</span>")
	affecting = null
```
Always null the references to things, to avoid hard deletes.
Where is this variable set? Register to its destruction as well and handle it.

IndieanaJones committed:
Update code/modules/antagonists/space_ninja/space_ninja.dm

Co-authored-by: Rohesie <rohesie@gmail.com>

IndieanaJones committed:
Update code/modules/antagonists/space_ninja/space_ninja.dm

Co-authored-by: Rohesie <rohesie@gmail.com>

IndieanaJones committed:
Update code/modules/events/space_ninja.dm

Co-authored-by: Rohesie <rohesie@gmail.com>

IndieanaJones committed:
Update code/modules/ninja/energy_katana.dm

Co-authored-by: Rohesie <rohesie@gmail.com>

IndieanaJones committed:
Update code/modules/ninja/ninja_explosive.dm

Co-authored-by: Rohesie <rohesie@gmail.com>

IndieanaJones committed:
Update code/modules/ninja/suit/ninja_equipment_actions/energy_net_nets.dm

Co-authored-by: Rohesie <rohesie@gmail.com>

IndieanaJones committed:
Remove Unnecessary Condition Checks

IndieanaJones committed:
I'll show you what A stands for!

PKPenguin321 commented:
Does the suit still have the secret hyper mode

PKPenguin321 commented:
Nvm it was removed years ago and I'm senile https://github.com/tgstation/tgstation/pull/8261

IndieanaJones commented:
How many people looked over this code and didn't think randomized suit stats was a stupid idea?  It boggles me that even a single person looked at that and just didn't do anything about it.

PKPenguin321 commented:
We have been very aware of ninja being shitcode for a very long time

IndieanaJones committed:
Turn some things into defines

IndieanaJones committed:
Remove is_ninja()

IndieanaJones committed:
Gives the SN roles on Antag Gain, not on event spawn

IndieanaJones committed:
Clean up mind/key code to be efficient and modern

IndieanaJones commented:
OK, reviews have been managed.  We're now using minds properly, and I've also set SN's spawn population to 20 from 15, since that's in-line with most midround antagonists.

IndieanaJones commented:
I have some sprite stuff on the works which I should have ready later today.  I'll then slap a [READY] on this and hopefully we can get approval for a testmerge going.

IndieanaJones committed:
Have Deactivated Suit Gloves use default black sprite

IndieanaJones committed:
Add the new clothing sprites

IndieanaJones committed:
Add new inventory sprites for alternate sprites

IndieanaJones committed:
Get Rid of These Again

IndieanaJones committed:
Re-add suit changes

IndieanaJones commented:
The changes are currently a bit borked from rebasing, will re-add the new sprites tomorrow.

ExcessiveUseOfCobblestone commented on code/modules/ninja/suit/gloves.dm:
```diff
@@ -1,41 +1,34 @@
 	var/candrain = FALSE
 	var/mindrain = 200
 	var/maxdrain = 400
+	var/cyborg_hijacks = 1
+	var/security_console_hacks = 1
+	var/communication_console_hacks = 1
+	var/door_hack_counter = 0
```
docs

ExcessiveUseOfCobblestone commented on code/modules/ninja/suit/ninja_equipment_actions/ninja_adrenaline.dm:
```diff
@@ -0,0 +1,45 @@
+  * radium by calling a different proc.
+  */
+/obj/item/clothing/suit/space/space_ninja/proc/ninjaboost()
+	if(!ninjacost(0,N_ADRENALINE))
+		var/mob/living/carbon/human/ninja = affecting
+		ninja.SetUnconscious(0)
+		ninja.SetStun(0)
+		ninja.SetKnockdown(0)
+		ninja.SetImmobilized(0)
+		ninja.SetParalyzed(0)
+		ninja.adjustStaminaLoss(-75)
+		ninja.stuttering = 0
+		ninja.update_mobility()
+		ninja.reagents.add_reagent(/datum/reagent/medicine/stimulants, 5)
+		ninja.say(pick("A CORNERED FOX IS MORE DANGEROUS THAN A JACKAL!","HURT ME MOOORRREEE!","IMPRESSIVE!"), forced = "ninjaboost")
+		a_boost = FALSE
+		to_chat(ninja, "<span class='notice'>You have used the adrenaline boost.</span>")
+		s_coold = 6
+		addtimer(CALLBACK(src, .proc/ninjaboost_after), 70)
```
```DM
if(thing)
   return
code

ExcessiveUseOfCobblestone commented on code/modules/ninja/suit/ninja_equipment_actions/ninja_cost_check.dm:
```diff
@@ -0,0 +1,27 @@
+  * Arguments:
+  * * cost - the energy cost of the ability
+  * * specificCheck - Determines if the check is a normal one, an adrenaline one, or a stealth cancel check.
+  */
```
document return value meaning

ExcessiveUseOfCobblestone commented on code/modules/ninja/suit/ninja_equipment_actions/ninja_cost_check.dm:
```diff
@@ -0,0 +1,27 @@
+  * * cost - the energy cost of the ability
+  * * specificCheck - Determines if the check is a normal one, an adrenaline one, or a stealth cancel check.
+  */
+/obj/item/clothing/suit/space/space_ninja/proc/ninjacost(cost = 0, specificCheck = 0)
```
are the return values boos

IndieanaJones replied:
No.

ExcessiveUseOfCobblestone commented on code/modules/ninja/suit/ninja_equipment_actions/ninja_empulse.dm:
```diff
@@ -0,0 +1,20 @@
+  * Proc called to allow the ninja to EMP the nearby area.  By default, costs 500E, which is half of the default battery's max charge.
+  * Also affects the ninja as well.
+  */
+/obj/item/clothing/suit/space/space_ninja/proc/ninjapulse()
+	if(!ninjacost(500,N_STEALTH_CANCEL))
+		var/mob/living/carbon/human/H = affecting
+		playsound(H.loc, 'sound/effects/empulse.ogg', 60, 2)
+		empulse(H, 4, 6) //Procs sure are nice. Slightly weaker than wizard's disable tch.
+		s_coold = 4
```
```DM
if(thing)
    return
code

ExcessiveUseOfCobblestone commented on code/modules/ninja/suit/ninja_equipment_actions/ninja_stars.dm:
```diff
@@ -0,0 +1,42 @@
+  * star doesn't do much up-front damage, but deals stamina damage
+  * as the target moves around, forcing a finish or flee scenario.
+  */
+/obj/item/clothing/suit/space/space_ninja/proc/ninjastar()
+	if(!ninjacost(10))
+		var/mob/living/carbon/human/ninja = affecting
+		var/obj/item/throwing_star/stamina/ninja/ninja_star = new(ninja)
+		if(ninja.put_in_hands(ninja_star))
+			to_chat(ninja, "<span class='notice'>A throwing star has been created in your hand!</span>")
+		else
+			qdel(ninja_star)
+			to_chat(ninja, "<span class='notice'>You can't create a throwing star, your hands are full!</span>")
+		ninja.throw_mode_on() //So they can quickly throw it.
```
```DM
if(thing)
    return
do

ExcessiveUseOfCobblestone commented on code/modules/ninja/suit/ninja_equipment_actions/ninja_stealth.dm:
```diff
@@ -0,0 +1,44 @@
+  *
+  * Called to cancel the stealth effect if it is ongoing.
+  * Does nothing otherwise.
+  */
```
doc return value meaning

ExcessiveUseOfCobblestone commented on code/modules/ninja/suit/suit.dm:
```diff
@@ -1,57 +1,69 @@
-
-/obj/item/clothing/suit/space/space_ninja/get_cell()
-	return cell
+/obj/item/clothing/suit/space/space_ninja/examine(mob/user)
+	. = ..()
+	if(s_initialized)
+		if(user == affecting)
+			. += "All systems operational. Current energy capacity: <B>[DisplayEnergy(cell.charge)]</B>.\n"+\
+			"The CLOAK-tech device is <B>[stealth?"active":"inactive"]</B>.\n"+\
+			"[a_boost?"An adrenaline boost is available to use.":"There is no adrenaline boost available.  Try refilling the suit with 20 units of radium."]"
```
```DM
. = ..()
if(!check)
   return
if(check2 != thing)
   return
code

ExcessiveUseOfCobblestone commented on code/modules/ninja/suit/suit.dm:
```diff
@@ -96,113 +108,146 @@ Contents:
-	a_boost = rand(1,7)
-
+/obj/item/clothing/suit/space/space_ninja/get_cell()
+	return cell
```
is this not default

ExcessiveUseOfCobblestone commented on code/modules/ninja/suit/suit.dm:
```diff
@@ -96,113 +108,146 @@ Contents:
+  * If the person in the suit isn't a ninja when this is called, this proc just gibs them instead.
+  * Arguments:
+  * * ninja - The person wearing the suit.
+  */
```
return value meaning

ExcessiveUseOfCobblestone commented on code/modules/events/space_ninja.dm:
```diff
@@ -0,0 +1,50 @@
+		return MAP_ERROR
+
+	//selecting a candidate player
+	var/list/candidates = get_candidates(ROLE_NINJA, null, ROLE_NINJA)
```
this is different than poll right?

IndieanaJones replied:
Yes, this is a specific method used for all midround antagonists.

ExcessiveUseOfCobblestone commented on code/modules/antagonists/space_ninja/space_ninja.dm:
```diff
@@ -0,0 +1,112 @@
+	survival.owner = owner
+	objectives += survival
+
+/proc/remove_ninja(mob/living/ninja)
```
🤢

ExcessiveUseOfCobblestone reviewed (commented):
i didnt look at deleted stuff and assumed if you moved it you are willing to doc it 😏

IndieanaJones committed:
Implement all the suggestions for the antag file

IndieanaJones committed:
And then everything else

IndieanaJones commented:
Alright, that handles all Cobby's review stuff and we've fixed the new sprites.  Speaking of the new sprites, let's talk about that real quick:

Space ninja now has two different outfit appearances they can select from when the suit is initializing.  You can choose between classic:
![image](https://user-images.githubusercontent.com/47086570/94621872-1c125e00-027f-11eb-9637-da06e21eb260.png)

And the New Age:
![image](https://user-images.githubusercontent.com/47086570/94621902-27658980-027f-11eb-9d56-5042192510c0.png)

As you can see, the katana has been updated between them both, and it looks much nicer now I think.  Big thanks to Azlan for the sprites!

IndieanaJones committed:
Updating the Weight for purpose of testmerge

IndieanaJones commented:
The change in weight isn't anything permanent, I was asked to up it for the eventual test merge just so admin intervention wasn't needed to make it spawn.

IndieanaJones commented:
Current thoughts of what I've seen during the test merge:
- Net nets to be buffed, isn't worth anything as it is right now (durability up and cost reduction)
- Give the katana better notifications for its amount of dashes left, maybe re-introduce the third dash
- Possibly reduce stealth cost
- Maybe more guidance on playing him stealthily
- The spider charge could use an additional effect, suggestion has been cast out that it might be used as a signal for something else to occur

Overall, the ninja nerfs have taken their toll and he's definitely much weaker now, though perhaps he's too weak as it stands.  He definitely can't fight a crowd by himself anymore which is good, but he could use buffs in terms of the net which doesn't serve much of a purpose right now and an easier time playing stealthily by reducing stealth cost.  Syndicate cyborgs were surprisingly not as awful as one might think they'd be including the assault cyborg, and each one provides a unique benefit to the ninja (assault having direct offensive power the ninja lacks, medic being able to heal the ninja while his suit is on along with being offensive itself, and sabotage being able to help ruin the stations infrastructure), the only thing being that the ninja cannot heal the assault or medical borg easily which helps to balance them out.  I also want to look into doing something more interesting with the spider charge, since if the game chooses for you to detonate in a boring location, it doesn't have much impact on the game.

davethwave commented:
One thing that I noticed with a bit of testing is that you can't put a PAI in your suit anymore with the rework.

IndieanaJones commented:
> One thing that I noticed with a bit of testing is that you can't put a PAI in your suit anymore with the rework.

This feature has been long gone before the rework ever happened.  The wiki is extremely out of date.

MoonFalcon commented:
I don't have any hard stats on this, but when I tested it charging took a really long time, even on the base standard battery. This was recharging through APCS, SMESs, and Solar wires. Each time you have to charge up its going to take you out of the game for about 5 minutes.

This adds up quickly and spending these periods standing in one spot charging is very boring. You could drag around a cell charger and carry around a shitty battery to swap and put in the charger, but that won't work if you teleport at any point and is very obtuse.

I believe you should either add self recharging or greatly increase the amount of power you get from things like wires, APCs, and SMESs. This would reduce the amount of very boring standing in place 'gameplay'

MoonFalcon commented:
Another concern I have is with the net. Why have a net that slows people down if you can't melee people through it? AFAIK you can only shoot people or alt click to hit people under the net sprite, you can't do ti directly. If the net is meant as a getaway it should have its power requirements (or cooldown if you go that route) massively reduced.

IndieanaJones commented:
The abilities have do put the suit in cooldown for a little bit, but none of them are very long.  I'll have a patch ready soon which will greatly ease up on the power requirements.

IndieanaJones committed:
Balance Updates

IndieanaJones committed:
Update the Objective Definitions

IndieanaJones commented:
Alright, this update is mainly focused on resolving the critique of the current ninja, which is mainly focused around buffing the ninja since the overwhelming answer that that they were too weak and that limiting some of the abilities felt restrictive and ultimately made the ninja feel not very effective.  A list of the changes:

**KATANA**
- No longer jaunts when attacking something on an adjacent tile
- Stored jaunts has been changed back from 2 to 3.  Note that the amount of time it takes to re-gain a jaunt is still longer than it was.

**ADRENALINE**
- Now removes 200 stamina damage, up from 75 stamina damage
- The amount of radium injected into the ninja afterwards is now 5 instead of 20.

**NINJA NET**
- Integrity buffed from 40 to 60
- Cost reduced from 80E to 40E

**SPIDER CHARGE**
- The explosion has been made stronger

**GLOVES**
- The syndicate cyborg interaction is no longer limited to one use.  You are only required to do this once in order to complete the objective.
- The security console interaction is no longer limited to one use.  You are only required to do this once in order to complete the objective.
- The communication console interaction no longer displays a fake midround antagonist announcement, but will instead either start the space pirate or swarmer event.  The time to hack the console has been upped to 30 seconds and the AI is now alerted once you begin.  This is still limited to one use.

**STEALTH**
- Cost greatly reduced (20 -> 4 energy expent per delta time)

IndieanaJones committed:
Remove Now Unreachable Cyborg Interaction Code

IndieanaJones committed:
Get That Shit Out of My Forest

IndieanaJones committed:
Re-Add Cowl Wearing Sprite

IndieanaJones committed:
Re-Add Cowl Inventory Sprite

Bawhoppen commented:
Maybe leave in stunning with ninja gloves? It was a nice homage to the original stungloves, gone, but not forgotten.

I don't think it was that useful luckily, so I don't feel it would be a balance mistake to leave it there if you chose to.... of course I understand removing it

SkeletalElite commented:
> I don't think it was that useful luckily, so I don't feel it would be a balance mistake to leave it there if you chose to.... of course I understand removing it

or just make them knockdown instead of hardstun

MoonFalcon commented:
Yup a knockdown with some burn damage would be great for the gloves. Maybe even chance to wound. How much damage is required to slow someone down?

IndieanaJones committed:
Adapt to the new thing?.thing changes

IndieanaJones committed:
Re-add New Katana Inhand Sprites

IndieanaJones commented:
I mean, sure, I wouldn't mind re-adding the carbon shock to the gloves as long as its a knockdown instead of an electrocute.  The original also had a 10% charge cost which seems balanced enough for this.

IndieanaJones committed:
Re-Add glove's shock but have it do knockdown instead

actions-user committed:
Automatic changelog compile [ci skip]

actions-user committed:
Automatic changelog compile [ci skip]

IndieanaJones committed:
Revert "Automatic changelog compile [ci skip]"

This reverts commit 48a63defa644ec9826ee5eb1736e5158d2a636d4.

IndieanaJones committed:
Revert "Automatic changelog compile [ci skip]"

This reverts commit cb68f4988edd9d841c4203eecee2067a49ce75c5.

IndieanaJones committed:
Revert "Revert "Automatic changelog compile [ci skip]""

This reverts commit e6e1f1ef53ee6611c334efcae6ca7c102edaf351.

IndieanaJones committed:
Revert "Revert "Automatic changelog compile [ci skip]""

This reverts commit 1782210527b9819772c0781122bec35b7979bf7d.

IndieanaJones committed:
Update changelog.html

actions-user committed:
Automatic changelog compile [ci skip]

ExcessiveUseOfCobblestone commented:
mfw you forcepush so i cant use the "review new commits" frick u !!!!

actions-user committed:
Automatic changelog compile [ci skip]

IndieanaJones commented:
![image](https://user-images.githubusercontent.com/47086570/96391281-2d67cf80-1186-11eb-90df-74988c96dd03.png)

IndieanaJones committed:
Get rid of the these stupid changelog changes

IndieanaJones committed:
Update sounds

IndieanaJones committed:
Re-add New Mask Sprite

IndieanaJones committed:
Re-update katana inventory sprite

IndieanaJones committed:
Revert Event Weight

IndieanaJones commented:
I'm not going to fix the conflicts until somebody actually looks at this or maybe a testmerge gets greenlit.

ExcessiveUseOfCobblestone commented:
if u forcepush again i will not be reviewing ur pr

IndieanaJones committed:
Update code/modules/ninja/suit/ninja_equipment_actions/energy_net_nets.dm

Co-authored-by: girl <11748095+ExcessiveUseOfCobblestone@users.noreply.github.com>

ExcessiveUseOfCobblestone commented:
fix the DM conflicts, let me review, fix DMI, merge

ping me along the process through discord, ive been bad about checking github recently.

ExcessiveUseOfCobblestone commented on code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm:
```diff
@@ -590,6 +590,42 @@
+	if(!spawn_locs.len)
+		message_admins("No valid spawn locations found in GLOB.xeno_spawn, aborting swarmer spawning...")
+		return MAP_ERROR
+	var/obj/structure/swarmer_beacon/new_beacon = new /obj/structure/swarmer_beacon(pick(spawn_locs))
```
`new(pick(spawn_locs))`

ExcessiveUseOfCobblestone commented:
ok fix any pending issues and ping me on discord

IndieanaJones committed:
Re-Add Dynamic Ninja

IndieanaJones committed:
Revert Changelog thing

IndieanaJones committed:
Update Mob Sprites

IndieanaJones committed:
Add files via upload
